Doc/reference/datamodel.rst

```diff
@@ -2330,20 +2381,28 @@ through the object's keys; for sequences, it should iterate through the values.
 
 .. method:: object.__getitem__(self, key)
 
-   Called to implement evaluation of ``self[key]``. For sequence types, the
-   accepted keys should be integers and slice objects.  Note that the special
-   interpretation of negative indexes (if the class wishes to emulate a sequence
-   type) is up to the :meth:`__getitem__` method. If *key* is of an inappropriate
-   type, :exc:`TypeError` may be raised; if of a value outside the set of indexes
-   for the sequence (after any special interpretation of negative values),
-   :exc:`IndexError` should be raised. For mapping types, if *key* is missing (not
-   in the container), :exc:`KeyError` should be raised.
+   Called to implement evaluation of ``self[key]``, which is translated by the
+   interpreter to ``type(self).__getitem__(self, key)``. For :term:`sequence`
+   types, the accepted keys should be integers and slice objects.  Note that
+   the special interpretation of negative indexes (if the class wishes to
+   emulate a :term:`sequence` type) is up to the :meth:`__getitem__` method. If
+   *key* is of an inappropriate type, :exc:`TypeError` may be raised; if of a
+   value outside the set of indexes for the sequence (after any special
+   interpretation of negative values), :exc:`IndexError` should be raised. For
+   :term:`mapping` types, if *key* is missing (not in the container),
+   :exc:`KeyError` should be raised.
 
    .. note::
 
       :keyword:`for` loops expect that an :exc:`IndexError` will be raised for illegal
       indexes to allow proper detection of the end of the sequence.
 
+   .. note::
+
+      When :ref:`subscripting<subscriptions>` a *class*, the special
+      classmethod :meth:`~object.__class_getitem__` may be called instead of
+      ``__getitem__()``.
```

Comment:
Since we worked so hard on the "`__class_getitem__` versus `__getitem__`" section, we might want to cross-reference it here.